src/poetry/utils/password_manager.py

```diff
@@ -46,9 +46,14 @@ def get_credential(
             return default
 
         import keyring
+        import keyring.errors
 
         for name in names:
-            credential = keyring.get_credential(name, username)
+            try:
+                credential = keyring.get_credential(name, username)
+            except keyring.errors.KeyringLocked:
```

Comment:
This is missing the https://github.com/python-poetry/poetry/issues/3012 error which is common for headless servers.

Reply:
I feel like no retry should happen, even with different service names, when getting a `KeyringLocked`, but I am not sure that behavior is appropriate for all the other `KeyringError` subclasses.